### PR TITLE
Abclattice

### PR DESCRIPTION
**Pull request by tuopuu**
Added ABCLattice abstract class and renamed id to index as was discussed in #39.

### CODE REVIEW

**Comment by itziakos** [comment]
I had a quick look:
- Please add the docstrings to the ABCLattice
- The Lattice class should either:
 -  subclass from `ABCLattice` or 
 -  subclass `object` and register itself as implementing the ABCLattice interface.

**Comment by itziakos** [comment]
I see that you fixed the second case

File: simphony/cuds/abstractlattice.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        ABCLattice ---> Common Base abstract class ("interface") for
+            the Lattice.
+"""
+
+from __future__ import print_function
+from abc import ABCMeta, abstractmethod
+
+
+class ABCLattice(object):
+    """Abstract base class for a Lattice item."""
```
Comment (nathanfranklin) [inline simphony/cuds/abstractlattice.py]: Should the `name` attribute be added to the docstring?

How about the other attributes used by Lattice ( `type, base_vect,  size, origin`)?  I assume they will be supported by FileLattice and any other classes implementing the ABCLattice interface.

Comment (itziakos) [inline simphony/cuds/abstractlattice.py]: Yes, I agree, the name is not part of all top level cuds containers

Comment (itziakos) [inline simphony/cuds/abstractlattice.py]: updated comment ``not -> now``

**Comment by nathanfranklin** [comment]
@tuopuu @kemattil   This looks good.  Can you address the remaining comments so that this can be merged?

**Comment by tuopuu** [comment]
I replaced abstractlattice.py with the version from @nathanfranklin 's abclattice_update branch, and added type, size, origin and base_vect attributes to docstrings (as @nathanfranklin suggested). These attributes are essential in defining a lattice and should be therefore always associated with the representing object.

**Comment by nathanfranklin** [comment]
@tuopuu  Thanks.  I will fix merge conflict in lattice.py and then merge

**Comment by tuopuu** [comment]
@nathanfranklin OK. What was causing a conflict if I may ask? Please note that there are some changes to lattice.py in this branch that should also be merged to master.

**Comment by nathanfranklin** [comment]
@tuopuu  there was a conflict with simphony/cuds/lattice.py when merging.  please review the commits then we we can merge this pr and close it.

**Comment by tuopuu** [comment]
lattice.py is fine now: I just changed size parameter to tuple as it is in ABCLattice description. In my opinion this merge is good to go.

**Pull request merged**